### PR TITLE
Try to make clipping parameters functional

### PR DESCRIPTION
**Pull request by ewels**
All of the trimming parameters were there in the pipeline, but they weren't actually being used 😱 Also they were being defined twice, which doesn't work as they are `params` and not regular variables.

Rewrote the code to apply it all in the config. Ugly, but hopefully functional..

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/methylseq/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/methylseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by ewels** [comment]
Tested manually a couple of examples, I think that the trimming presets are now working properly.

**Comment by ewels** [comment]
@Emiller88 for tests, the kind of manual thing I was doing was this:

```bash
nextflow run ../ -profile test,docker --outdir results --em_seq --clip_r1 2  
```

Then checking the `.command.sh` in that work dir to check that the trimming parameters are there in the command:

```bash
#!/bin/bash -euo pipefail
[ ! -f  SRR389222_sub1.fastq.gz ] && ln -s SRR389222_sub1.fastq.gz SRR389222_sub1.fastq.gz
trim_galore \
    --fastqc   --clip_r1 2  --three_prime_clip_r1 10 \
    --cores 1 \
    --gzip \
    SRR389222_sub1.fastq.gz

cat <<-END_VERSIONS > versions.yml
"NFCORE_METHYLSEQ:METHYLSEQ:TRIMGALORE":
    trimgalore: $(echo $(trim_galore --version 2>&1) | sed 's/^.*version //; s/Last.*$//')
    cutadapt: $(cutadapt --version)
END_VERSIONS
```

Here you can see:

- `--three_prime_clip_r1 10` from the `--em_seq` preset
- `--clip_r1 2` correctly overriding from `--clip_r1`

File: tests/clipping/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,25 @@
+nextflow_pipeline {
+
+    name "Clipping"
+    script "main.nf"
+
+    test("Should run with clipping params") {
+        when {
+            params {
+                aligner = "bismark"
```
Comment (ewels) [inline tests/clipping/main.nf.test:9]: This is the default, so probably not needed..

Comment (ewels) [reply tests/clipping/main.nf.test]: Looking at this fresh I'm going back on my previous opinion, as I think it makes it clearer to read even if it's not needed 😅  (now I have several of these files open at once)

**Review by FelixKrueger — APPROVED** [review]
LGTP.

**Pull request merged**